channel.go

```diff
@@ -200,11 +201,16 @@ func (c *serialChannel) listen() (net.Listener, error) {
 	if err != nil {
 		return nil, err
 	}
+	c.waitCh = session.CloseChan()
 
 	return session, nil
 }
 
 func (c *serialChannel) teardown() error {
+	// wait for the session to be fully shutdown first
+	if c.waitCh != nil {
+		<-c.waitCh
```

Comment:
You should use a go `select:` here, and timeout after some time, otherwise this might result into a deadlock.

Reply:
hmm, good point. I'll get to it later today.